### DISCUSSION TITLE
Does emmeans work with delta models?

### DISCUSSION THREAD

**Discussion by sutjo** [sdmTMB/sdmTMB#247] [Q&A] [answered]
Good morning. I would like to use emmeans with delta models, but it does not seem to be supported. Would modelling the delta components separately and then using emmeans be an acceptable workaround? I.e. Modeling the binomial and the positive catch components separately, then applying emmeans to each of the two models. 

Here is an example of my code:
emmeans::emmeans(fit.delta, ~ predictor)
#result = Error in vcov.sdmTMB(object) : nrow(b) == nrow(vc) is not TRUE

emmeans::emmeans(fit.binomial, ~ predictor)
#result = estimates marginal means

emmeans::emmeans(fit.pos.catch ~ predictor)
#result = estimates marginal means; note that the pos.catch data set does not include zero catches

**Comment by seananderson** [comment, marked as answer]
Yes, you could usually fit the components separately. I'll add an issue and hope to get to this soon.

Fitting them separately should work in all cases with the following exceptions:
- if you are using the 'Poisson-link' families
- if you are sharing parameters between the models; this won't happen without a custom 'map' argument you've written with the exception of anisotropy, which by default is shared between the components.

In the meantime, I added a more helpful error message.

**Comment by seananderson** [comment]
Sorry this took so long, but this is now fixed in the GitHub package version:

``` r
library(sdmTMB)

fit_delta <- sdmTMB(
  density ~ as.factor(year),
  data = pcod_2011,
  spatial = "off",
  family = delta_gamma()
)

emmeans::emmeans(fit_delta, ~ year, model = 1)
#>  year  emmean    SE  df lower.CL upper.CL
#>  2011 -0.4455 0.129 960  -0.6994   -0.192
#>  2013  0.2344 0.130 960  -0.0207    0.489
#>  2015  0.0841 0.130 960  -0.1706    0.339
#>  2017 -0.5108 0.133 960  -0.7725   -0.249
#> 
#> Results are given on the logit (not the response) scale. 
#> Confidence level used: 0.95
emmeans::emmeans(fit_delta, ~ year, model = 2)
#>  year emmean    SE  df lower.CL upper.CL
#>  2011   4.73 0.126 960     4.48     4.97
#>  2013   4.28 0.108 960     4.06     4.49
#>  2015   4.63 0.112 960     4.41     4.85
#>  2017   4.21 0.132 960     3.95     4.47
#> 
#> Results are given on the log (not the response) scale. 
#> Confidence level used: 0.95
```

<sup>Created on 2025-10-18 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>
